dxr/plugins/python/indexers.py

```diff
@@ -25,14 +26,27 @@
 }
 
 
+class _FileToIgnore(object):
+    """A file that we don't want to bother indexing, usually due to
+    syntax errors.
+
+    """
+    def is_interesting(self):
+        return False
+FILE_TO_IGNORE = _FileToIgnore()
+
+
 class TreeToIndex(TreeToIndexBase):
     def __init__(self, *args, **kwargs):
         super(TreeToIndex, self).__init__(*args, **kwargs)
 
-        self.python_path = getattr(self.tree, 'plugin_python_path', self.tree.source_folder)
+        self.python_path = self.plugin_config.python_path
```

Comment:
Unfortunately, this loses the defaulting to tree.source_folder. I'd like the config system to provide a mechanism for dependent defaults ultimately, but, for now, the best solution is probably to make it `Optional` in the config_schema, test for its presence here, and fall back to the default if absent.